lib/auth/AuthRepository.dart

```diff
@@ -16,14 +17,18 @@ class AuthRepository {
     _auth = FirebaseAuth.instance;
   }
 
-  Future<FirebaseUser> signInUsingGoogle() async {
+  Future<FirebaseUser> signInUsingGoogle(FlutterSecureStorage _secureStorage) async {
```

Comment:
Instead of passing the instance of the secure storage as an argument, it would be better if the instance is created when the repository is instantiated.
Although your implementation is also correct, according to the separation of concerns principle, the view should not be aware of where the data comes from. Hence, it should not depend on the fact that we are storing key-value pairs in Secure storage or in Shared Preferences